From 82b5d4028885d75adba9b66d43aeb11592e64914 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Wed, 13 Oct 2021 18:16:55 -0700
Subject: [PATCH] fix runtime with wolfSSL and fastmath

wolfSSL's fastmath support requires options.h to be included before
anything else. Otherwise bad codepaths get taken and a crash occurs
during DH initialization.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 libtransmission/crypto-utils-cyassl.c | 1 +
 1 file changed, 1 insertion(+)

--- a/libtransmission/crypto-utils-cyassl.c
+++ b/libtransmission/crypto-utils-cyassl.c
@@ -19,6 +19,7 @@
 #define API_VERSION_HEX LIBCYASSL_VERSION_HEX
 #endif
 
+#include API_HEADER(options.h)
 #include API_HEADER_CRYPT(arc4.h)
 #include API_HEADER_CRYPT(dh.h)
 #include API_HEADER_CRYPT(error-crypt.h)
